### ISSUE TITLE
No clear directions for integrating WIL into CMake projects

### ISSUE THREAD

**Issue by CJKay** [microsoft/wil#118]
Generally speaking, CMake libraries that are intended for use elsewhere will install an exports file - usually in `${INSTALL_DIR}/cmake/<project>Config.cmake` - in order to allow third party CMake projects to locate the library with `find_package(<project>)`. WIL does not appear to provide this CMake configuration file, and so it is unclear how the user is expected to integrate it into other CMake projects, and means that CMake projects that would otherwise use `find_package()` to pull in their dependencies now need to provide custom logic for WIL.

Now, it appears to me that CMake is not strictly used to package the library itself, but rather to do the packaging and testing tasks. I think the library would be well served with an [exported](https://visualgdb.com/tutorials/linux/cmake/find_package/) [interface library target](https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries) which we could then pull in much more easily.

**Comment by dunhor** [comment]
Correct. The use of CMake here is predominantly for building the tests. Feel free to issue a PR if the scenario is important to you.

**Comment by CookiePLMonster** [comment]
Allow me to second this - I had some trouble integrating WIL in Dolphin's cmake: https://github.com/dolphin-emu/dolphin/pull/8427

**Comment by sylveon** [comment]
You might be interested in #123
